contracts/cw2981-royalties/src/msg.rs

```diff
@@ -0,0 +1,139 @@
+use cosmwasm_std::Uint128;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cw721_base::msg::QueryMsg as CW721QueryMsg;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum Cw2981QueryMsg {
```

Comment:
I wish there were a better way to extend enums.
But, alas, there currently is not.

Reply:
Haven't tested this with a production build yet, but you can extend an enum using serde `untagged` with an extra variant for the extended functionality. So this has the same effect as `flatten`. 

```rs
#[serde(untagged)]
pub enum QueryMsg {
    Base(cw721_base::QueryMsg),
    Extended(ExtendedQueryMsg),
}
```

Reply:
Please try that was `cargo wasm` and let me know.
I bet it hits a limit of serde-json-wasm, but if not, that is a great work-around

Reply:
I think this may not work as expected as it will only affect the serialization of the enum to JSON. However, when dealing with a contract calling another contract, I believe you will still need the "tagged" compound enum. Need to test this, but I think if a second contract tried to interact with this contract believing it was an unextended cw721_base it would fail unless for some reason cosmwasm leverages Serde for internal calls. 

TLDR this might work from a client's perspective, but I don't think it would work from a second contract's perspective.

Reply:
I will give it a try and see if the workaround works 🤔

Reply:
> 

This is correct. :cry:

Yeah, there really needs to be a macro for this.

Reply:
Ah yeah, that makes sense now I've taken the time to actually parse what @ben2x4 was getting at 😅

Reply:
@ethanfrey It didn't work. Seemed too good to be true :) Will have to dive into serde-json-wasm one of these days and figure it out.

@ben2x4 The tagged unextended and untagged extended should be represented exactly the same during transport and survive serialization and deserialization. That was the hope anyway. Have to dig into the guts of the expansive Wasmverse to figure this out.